R/object.R

```diff
@@ -0,0 +1,123 @@
+#' nimue simulation plot
+#'
+#' @param x An squire_simulation object
+#' @param replicates Plot replicates
+#' @param summarise Logical, add summary line
+#' @param ci logical add confidence interval ribbon
+#' @param q Quantiles for upper and lower of interval ribbon
+#' @param var_select Vector of variable names to plot (default is all)
+#' @param summary_f Function to summarise each compartment
+#'   passed to the \code{fun} argument of \code{\link[ggplot2]{stat_summary}}
+#' @param x_var X variable to use for plotting (default is \code{"t"},
+#'   but can be set to, \code{"date"}, if \code{date_0} provided), which will
+#'   cause the date to be plotted rather than time.
+#' @param particle_fit If the squire_simulation provided is the result of
+#'   running the particle filter, do we want to just plot the fit. Default =
+#'   FALSE
+#' @param date_0 Date of time 0 (e.g. "2020-03-01"), if specified a date column
+#'   will be added
+#' @param ... additional arguments affecting the plot produced.
+#'
+#' @export
+#'
+plot.nimue_simulation <- function(x,
+                                  var_select = NULL,
+                                  replicates = FALSE,
+                                  summarise = TRUE,
+                                  ci = TRUE,
+                                  q = c(0.025, 0.975),
+                                  summary_f = mean,
+                                  x_var = "t",
+                                  date_0 = NULL,
+                                  particle_fit = FALSE,
+                                  ...) {
+
+  # work out what compartments are being plotted
+  compartments = c("S", "E",
+                   "IMild", "ICase", "IICU", "IHospital",
+                   "IRec", "R", "D")
+  summaries = c("N",
+                "hospitalisations",
+                "hospital_demand","hospital_occupancy",
+                "ICU_demand", "ICU_occupancy",
+                "vaccines", "unvaccinated", "vaccinated", "priorvaccinated",
+                "infections", "deaths")
+
+  comps <- var_select[var_select %in% compartments]
+  summs <- var_select[var_select %in% summaries]
+
+  # get the compartments requried
+  pd <- do.call(rbind, lapply(seq_len(dim(x$output)[3]), function(i) {
+    format(x, compartments = comps, summaries = summs, replicate = i)
+    })) %>%
+    dplyr::rename(y = .data$value)
+
+  # replacing time with date if date_0 is provided
+  if(!is.null(date_0)){
+    assert_date(date_0)
+    pd$date <- as.Date(pd$t + as.Date(date_0),
+                        format = "%Y-%m-%d")
+  }
+
+  # make the x label be the x axis requested
+  pd <- pd %>% dplyr::mutate(x = .data[[x_var]])
+
+
```

Comment:
Most of this I think would be handled by nimue::format

Reply:
So format is doing most of the work here on line 51. The filler above is so that you just have one argument for the vars and then just getting from format the variables that are requested rather than all of them. And then the lapply loop is because format is set up to get a specific replicate. Because the squire fitting infrastructure will return runs that have more than 1 replicate, i.e. multiple draws from posterior, then just need to handle those cases when we have multiple runs.